src/analysis/normed_space/ordered.lean

```diff
@@ -48,3 +67,146 @@ instance : normed_linear_ordered_field ℚ :=
 noncomputable
 instance : normed_linear_ordered_field ℝ :=
 ⟨dist_eq_norm, normed_field.norm_mul⟩
+
+/-!
+### Normed lattice orderd groups
+
+Motivated by the theory of Banach Lattices, this section introduces normed lattice ordered groups.
+-/
+
+local notation `|`a`|` := abs a
+
+/--
+Let `α` be a normed commutative group equipped with a partial order covariant with addition, with
+respect which `α` forms a lattice. Suppose that `α` is *solid*, that is to say, for `a` and `b` in
+`α`, with absolute values `|a|` and `|b|` respectively, `|a| ≤ |b|` implies `∥a∥ ≤ ∥b∥`. Then `α` is
+said to be a normed lattice ordered group.
+-/
+class normed_lattice_add_comm_group (α : Type*)
+  extends normed_group α, lattice α :=
+(add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
+(solid : ∀ a b : α, |a| ≤ |b| → ∥a∥ ≤ ∥b∥)
+
+/--
+A normed lattice ordered group is a lattice ordered group
+-/
+instance normed_lattice_add_comm_group_to_covariant_class {α : Type*}
+  [h : normed_lattice_add_comm_group α] : covariant_class α α (+) (≤) :=
```

Comment:
You don't need to name instance variables, except in unusual circumstances where you need to pass it directly.

Reply:
Somewhat to my surprise, I found it necessary to pass `normed_lattice_add_comm_group_to_covariant_class` twice in the proof of `normed_lattice_add_comm_group_has_continuous_inf`. Perhaps there is a better way of doing that proof?